src/core/batch/batch.ts

```diff
@@ -21,13 +24,34 @@ export type BatchFileObject = string;
 
 /**
  * createFile() takes a series of DSNP messages and returns a Batch file
- * object for publishing. This method is not yet implemented.
+ * object for publishing.
  *
- * @param events  An array of DSNPEvents to include in the batch file
- * @returns         A batch file object
+ * @param messages An array of DSNPMessage to include in the batch file
+ * @returns         A URL
  */
-export const createFile = (_events: DSNPType[]): BatchFileObject => {
-  throw NotImplementedError;
+export const createFile = async (targetPath: string, messages: DSNPMessage[]): Promise<URL | null> => {
```

Comment:
is there a strong reason to return null here? could we check for null and throw instead?

Reply:
I was not sure if we should. I will update it to throw.